src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/Cache.cs

```diff
@@ -0,0 +1,155 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectExplorer
+{
+    /// <summary>
+    /// A dictionary-based Cache that supports quick lookups for common objects. This is compatible with
+    /// <see cref="IDictionary"/> but note that <see cref="IDictionary.Remove(object)"/> has limited support - it
+    /// will effectively increase the cache size slightly since removing from the queue would be inefficient due to the
+    /// need to rebuild the recently used queue.
+    /// </summary>
+    /// <typeparam name="TKey">Key type</typeparam>
+    /// <typeparam name="TValue">Value type</typeparam>
+    internal class Cache<TKey, TValue> : IDictionary<TKey, TValue>
```

Comment:
Minor: I don't think we're currently using this. We will however need it in the future, just pointing this out for others. We can consider removing if needed.